lib/multigrid.cpp

```diff
@@ -599,22 +609,28 @@ namespace quda {
     transfer->R(*x_coarse, *tmp2);
     (*param_coarse->matResidual)(*r_coarse, *tmp_coarse);
 
-
 #if 0 // enable to print out emulated and actual coarse-grid operator vectors for debugging
-    if (getVerbosity() >= QUDA_VERBOSE) printfQuda("emulated\n");
-    for (int x=0; x<x_coarse->Volume(); x++) tmp1->PrintVector(x);
+    setOutputPrefix("");
 
-    if (getVerbosity() >= QUDA_VERBOSE) printfQuda("actual\n");
-    for (int x=0; x<r_coarse->Volume(); x++) tmp2->PrintVector(x);
-#endif
+    for (int i=0; i<comm_rank(); i++) { // this ensures that we print each rank in order
+      if (i==comm_rank()) {
+        if (getVerbosity() >= QUDA_VERBOSE) printfQuda("emulated\n");
+        for (int x=0; x<x_coarse->Volume(); x++) tmp1->PrintVector(x);
 
-    if (getVerbosity() >= QUDA_VERBOSE) printfQuda("Vector norms Emulated=%e Native=%e ", norm2(*x_coarse), norm2(*r_coarse));
+        if (getVerbosity() >= QUDA_VERBOSE) printfQuda("actual\n");
+        for (int x=0; x<r_coarse->Volume(); x++) tmp2->PrintVector(x);
+      }
+      comm_barrier();
+    }
+    setOutputPrefix(prefix);
+#endif
```

Comment:
Perhaps we should have two versions of `printVector`: one that takes a checkerboard index and prints the values for just this site, and then one that takes zero arguments and prints all of the sites using this pattern (i.e., loops over comm ranks, inner loop over all checkerboard sites). We could also give it a different name to avoid confusion. (`printEntireVector`?)

Reply:
Definitely for a subsequent pull.  @weinbe2 can you add an issue for this?

Reply:
Done, #807